### PR TITLE
Child Process Exit Code

### LINKED ISSUE #6342: make the types match of std.process.exit and std.ChildProcess exit code
In the self hosted compiler I have this code:

```zig
const child = try std.ChildProcess.init(argv.items, arena);
defer child.deinit();

if (comp.clang_passthrough_mode) {
    child.stdin_behavior = .Inherit;
    child.stdout_behavior = .Inherit;
    child.stderr_behavior = .Inherit;

    const term = child.spawnAndWait() catch |err| {
        return comp.failCObj(c_object, "unable to spawn {}: {}", .{ argv.items[0], @errorName(e
r) });
    };
    switch (term) {
        .Exited => |code| {
            if (code != 0) {
                // TODO make std.process.exit and std.ChildProcess exit code have the same type
                // and forward it here. Currently it is u32 vs u8.
                std.process.exit(1);
            }
        },
        else => std.process.exit(1),
    }
```

I wanted to do `std.process.exit(code)` which should in theory work but the types don't match.

### PR DESCRIPTION
**Pull request by gsquire**
Fixes #6342

### CODE REVIEW

**Comment by daurnimator** [comment]
I think the more correct fix would be for `WEXITSTATUS` to return a u8?

**Comment by gsquire** [comment]
So for each OS in [this search](https://github.com/ziglang/zig/search?p=2&q=WEXITSTATUS), you're suggesting we change the return type to a `u8`?

**Comment by daurnimator** [comment]
> So for each OS in [this search](https://github.com/ziglang/zig/search?p=2&q=WEXITSTATUS), you're suggesting we change the return type to a `u8`?

yes.
If that's not correct, then what's in this PR wouldn't be correct either.

File: lib/std/os/bits/netbsd.zig [hunk 1]
```diff
@@ -836,15 +836,13 @@ pub const ucontext_t = extern struct {
...
-    __pad: [
-        switch (builtin.arch) {
-            .i386 => 4,
-            .mips, .mipsel, .mips64, .mips64el => 14,
-            .arm, .armeb, .thumb, .thumbeb => 1,
-            .sparc, .sparcel, .sparcv9 => if (@sizeOf(usize) == 4) 43 else 8,
-            else => 0,
-        }
-    ]u32,
+    __pad: [switch (builtin.arch) {
```
Comment (gsquire) [inline lib/std/os/bits/netbsd.zig:839]: This was from my editor auto-formatting via `zig fmt`.

File: lib/std/child_process.zig [hunk 2]
```diff
@@ -324,7 +324,7 @@ pub const ChildProcess = struct {
             if (windows.kernel32.GetExitCodeProcess(self.handle, &exit_code) == 0) {
                 break :x Term{ .Unknown = 0 };
             } else {
-                break :x Term{ .Exited = exit_code };
+                break :x Term{ .Exited = @truncate(u8, exit_code) };
```
Comment (gsquire) [inline lib/std/child_process.zig:327]: Windows seems to handle this differently which isn't terribly surprising.

**Pull request merged**